src/google_cloud_debugger/google_cloud_debugger_lib/dbg_object_factory.cc

```diff
@@ -425,6 +421,119 @@ HRESULT DbgObjectFactory::EvaluateAndCreateDbgObject(
+      uintptr_t *result = (uintptr_t *)literal_value;
+      *dbg_object =
+          std::unique_ptr<DbgObject>(new DbgPrimitive<uintptr_t>(*result));
+      *numerical_value = *result;
+      return S_OK;
+    }
+    case CorElementType::ELEMENT_TYPE_I1: {
+      int8_t *result = (int8_t *)literal_value;
+      *dbg_object =
+          std::unique_ptr<DbgObject>(new DbgPrimitive<int8_t>(*result));
+      *numerical_value = *result;
+      return S_OK;
+    }
+    case CorElementType::ELEMENT_TYPE_U1: {
+      uint8_t *result = (uint8_t *)literal_value;
+      *dbg_object =
+          std::unique_ptr<DbgObject>(new DbgPrimitive<uint8_t>(*result));
+      *numerical_value = *result;
+      return S_OK;
+    }
+    case CorElementType::ELEMENT_TYPE_I2: {
+      int16_t *result = (int16_t *)literal_value;
+      *dbg_object =
+          std::unique_ptr<DbgObject>(new DbgPrimitive<int16_t>(*result));
+      *numerical_value = *result;
+      return S_OK;
+    }
+    case CorElementType::ELEMENT_TYPE_U2: {
+      uint16_t *result = (uint16_t *)literal_value;
+      *dbg_object =
+          std::unique_ptr<DbgObject>(new DbgPrimitive<uint16_t>(*result));
+      *numerical_value = *result;
+      return S_OK;
+    }
+    case CorElementType::ELEMENT_TYPE_I4: {
+      int32_t *result = (int32_t *)literal_value;
+      *dbg_object =
+          std::unique_ptr<DbgObject>(new DbgPrimitive<int32_t>(*result));
+      *numerical_value = *result;
+      return S_OK;
+    }
+    case CorElementType::ELEMENT_TYPE_U4: {
+      uint32_t *result = (uint32_t *)literal_value;
+      *dbg_object =
+          std::unique_ptr<DbgObject>(new DbgPrimitive<uint32_t>(*result));
+      *numerical_value = *result;
+      return S_OK;
+    }
+    case CorElementType::ELEMENT_TYPE_I8: {
+      int64_t *result = (int64_t *)literal_value;
+      *dbg_object =
+          std::unique_ptr<DbgObject>(new DbgPrimitive<int64_t>(*result));
+      *numerical_value = *result;
+      return S_OK;
+    }
+    case CorElementType::ELEMENT_TYPE_U8: {
+      uint64_t *result = (uint64_t *)literal_value;
+      *dbg_object =
+          std::unique_ptr<DbgObject>(new DbgPrimitive<uint64_t>(*result));
+      *numerical_value = *result;
+      return S_OK;
+    }
+    case CorElementType::ELEMENT_TYPE_R4: {
+      float *result = (float *)literal_value;
+      *dbg_object =
+          std::unique_ptr<DbgObject>(new DbgPrimitive<float>(*result));
+      return S_OK;
+    }
+    case CorElementType::ELEMENT_TYPE_R8: {
+      double *result = (double *)literal_value;
+      *dbg_object =
+          std::unique_ptr<DbgObject>(new DbgPrimitive<double>(*result));
+      return S_OK;
+    }
+    case CorElementType::ELEMENT_TYPE_STRING: {
+      WCHAR *string_ptr = (WCHAR *)literal_value;
+      std::vector<WCHAR> wchar_string_result(string_ptr,
+                                             string_ptr + literal_value_len);
+      wchar_string_result.push_back('\0');
```

Comment:
I think this will work but shouldn't this be L'\0' as we need to terminating bytes?